src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -95,73 +97,143 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+        email: row[hotInstanceColHeaders.indexOf(this.colHeaders[3])],
+        comments: row[hotInstanceColHeaders.indexOf(this.colHeaders[4])],
+      };
+      studentsEnrollRequest.studentEnrollRequests.push(request);
+    }
+
+    this.studentService.enrollStudents(this.courseid, studentsEnrollRequest).subscribe((resp: Students) => {
+      const enrolledStudents: Student[] = resp.students;
+      this.showEnrollResults = true;
+      this.statusMessage.pop(); // removes any existing error status message
+      this.statusMessageService.showSuccessMessage('Enrollment successful. Summary given below.');
+      this.enrollResultPanelList =
+          this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents,
+              studentsEnrollRequest.studentEnrollRequests);
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessage.pop(); // removes any existing error status message
+      this.statusMessageService.showErrorMessage(resp.error.message);
+    });
+    this.studentService.getStudentsFromCourse(this.courseid).subscribe((resp: Students) => {
+      this.existingStudents = resp.students;
+    });
+  }
+
+  private populateEnrollResultPanelList(existingStudents: Student[], enrolledStudents: Student[],
+                                         enrollRequests: StudentEnrollRequest[]): EnrollResultPanel[] {
+
+    const panels: EnrollResultPanel[] = [];
+    const studentLists: Student[][] = [];
+
+    for (const _ of Object.values(EnrollStatus).filter((value: EnrollStatus) => typeof value === 'string')) {
+      studentLists.push([]);
+    }
+
+    // Identify students not in the enroll list.
+    for (const existingStudent of existingStudents) {
+      const enrolledStudent: Student | undefined = enrolledStudents.find((student: Student) => {
+        return student.email === existingStudent.email;
+      });
+      if (enrolledStudent === undefined) {
+        studentLists[EnrollStatus.UNMODIFIED].push(existingStudent);
+      }
+    }
+
+    // Identify new students, modified students, and modified without any change students.
+    for (const enrolledStudent of enrolledStudents) {
+      const unchangedStudent: Student | undefined = existingStudents.find((student: Student) => {
+        return this.isSameEnrollInformation(student, enrolledStudent);
+      });
+      const modifiedStudent: Student | undefined = existingStudents.find((student: Student) => {
+        return student.email === enrolledStudent.email;
+      });
+      if (unchangedStudent !== undefined) {
+        studentLists[EnrollStatus.MODIFIEDUNCHANGED].push(enrolledStudent);
+      } else if (unchangedStudent === undefined && modifiedStudent !== undefined) {
+        studentLists[EnrollStatus.MODIFIED].push(enrolledStudent);
+      } else if (unchangedStudent === undefined && modifiedStudent === undefined) {
+        studentLists[EnrollStatus.NEW].push(enrolledStudent);
+      }
+    }
+
+    // Identify unsuccessfully enrolled students.
```

Comment:
<s>Identify unsuccessfully enrolled students.</s> 

Identify students that failed to enroll.